en/tutorials-and-examples/cms/authentication.rst

```diff
@@ -1,8 +1,142 @@
+get started, we'll add the :doc:`AuthComponent
+</controllers/components/authentication>` to our application. We'll want the
+create, update and delete methods to require authentication, so we'll add
+AuthComponent in our AppController::
+
+    // In src/Controller/AppController.php
+    namespace App\Controller;
+
+    use Cake\Controller\Controller;
+
+    class AppController extends Controller
+    {
+        public function initialize()
+        {
+            $this->loadComponent('Flash');
+            $this->loadComponent('Auth', [
+                'authenticate' => [
+                    'Form' => [
+                        'fields' => [
+                            'username' => 'email',
+                            'password' => 'password'
+                        ]
+                    ]
+                ],
+                'loginAction' => [
+                    'controller' => 'Users',
+                    'action' => 'login'
+                ],
+                 // If unauthorized, return them to page they were just on
+                'unauthorizedRedirect' => $this->referer()
+            ]);
+
+            // Allow the display action so our pages controller
+            // continues to work. Also enable the read only actions.
+            $this->Auth->allow(['display', 'view', 'index']);
+        }
+    }
+
+We've just told CakePHP that we want to load the ``Flash`` and ``Auth``
+components. In addition, we've customized the configuration of AuthComponent, as
+our users table uses ``email`` as the username. Now, if you go to any URL you'll
+be redirected to **/users/login**, which will show an error page as we have
+not written that code yet. So let's create the login action::
+
+    // In src/Controller/UsersController.php
+    public function login()
+    {
+        if ($this->request->is('post')) {
+            $user = $this->Auth->identify();
+            if ($user) {
+                $this->Auth->setUser($user);
+                return $this->redirect($this->Auth->redirectUrl());
+            }
+            $this->Flash->error('Your username or password is incorrect.');
+        }
+    }
+
+And in **src/Template/Users/login.ctp** add the following::
+
+    <h1>Login</h1>
+    <?= $this->Form->create() ?>
+    <?= $this->Form->control('email') ?>
+    <?= $this->Form->control('password') ?>
+    <?= $this->Form->button('Login') ?>
+    <?= $this->Form->end() ?>
+
+.. note::
+
+   The ``control()`` method is available since 3.4. For prior versions you can
+   use the ``input()`` method instead.
+
+Now that we have a simple login form, we should be able to log in with one of
+the users that has a hashed password.
+
+.. note::
+
+    If none of your users have hashed passwords, comment the
+    ``loadComponent('Auth')`` line. Then go and edit the user,
+    saving a new password for them.
+
+You should now be able to log in. If not, make sure you are using a user that
+has a hashed password.
```

Comment:
Maybe a new para:

"Try it out! Before logging in, visit \*\*/articles/add\*\*. Since this action is not allowed, you will be redirected to the login page. After logging in successfully, CakePHP will automatically redirect you back to \*\*/articles/add\*\*."